File: 백준/Silver/2839. 설탕 배달/설탕 배달.cc
#include "iostream"

using namespace std;

int main() {
    int n;
    cin >> n;
    int answer = 0;
    if (n % 5 == 0) {
        answer = n / 5;
        cout << answer << "\n";
    } else {
        while (n > 0) {
            n -= 3;
            answer++;
            if (n % 5 == 0) {
                answer += n/5;
                cout << answer << "\n";
                break;
            } else if (n == 1 || n == 2) {
                cout << -1 << "\n";
                break;
            } else if (n == 0) {
                cout << answer << "\n";
                break;
            }
        }
    }

    return 0;
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int answer = 0;
        if (N % 5 == 0) {
            System.out.println(N / 5);
        } else {
            while (N > 0) {
                answer++;
                N -= 3;
                if (N % 5 == 0) {
                    answer += N / 5;
                    System.out.println(answer);
                    break;
                } 
                if (N < 3) {
                    System.out.println(-1);
                    break;
                }
            }
        }
    }
}
